### PR TITLE
raftstore-v2: optimize RocksDB config

### PR DESCRIPTION
**Pull request by tabokie**
### What is changed and how it works?

Issue Number: Ref #12842 

What's Changed:

```commit-message
- Use latest data format
- Enable ribbon filter for bottommost level
- Limit compaction concurrency for defaultcf and writecf to 1, auto adjust when pending bytes is high
- Set max-total-wal-size to 1 for v2
```

### Related changes

https://github.com/tikv/rust-rocksdb/pull/754

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by tonyxuqqi** [comment]
Can you paste some benchmark results on the new change?

**Review by tonyxuqqi — APPROVED** [review]
LGTM, but in general needs some test (like benchbot) to verify that

**Comment by tabokie** [comment]
Benchbot XL 15m: oltp_insert improved by 3%, no difference for oltp_read_write and oltp_update_non_index

File: components/server/src/common.rs [hunk 1]
```diff
@@ -550,15 +558,43 @@ impl EnginesResourceInfo {
...
+                    };
+                    let cf = if i == 1 {
+                        CF_DEFAULT
+                    } else {
+                        CF_WRITE
+                    };
+                    self.tablet_registry.for_each_opened_tablet(|_, cache|
+                        if let Some(latest) = cache.latest() {
+                            let opts = latest.get_options_cf(cf).unwrap();
+                            if let Err(e) = opts.set_max_compactions(base + delta) {
```
Comment (SpadeA-Tang) [inline components/server/src/common.rs:590]: Can this go higher unlimitely?

**Review by SpadeA-Tang — APPROVED** [review]

**Comment by tonyxuqqi** [comment]
Ribbon filter should mitigate #14496 a bit.

**Pull request merged**